pkg/controller/remotemachineset/azureactuator.go

```diff
@@ -0,0 +1,115 @@
+	installertypes "github.com/openshift/installer/pkg/types"
+	installertypesazure "github.com/openshift/installer/pkg/types/azure"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/azureclient"
+)
+
+// AzureActuator encapsulates the pieces necessary to be able to generate
+// a list of MachineSets to sync to the remote cluster.
+type AzureActuator struct {
+	client azureclient.Client
+	logger log.FieldLogger
+}
+
+var _ Actuator = &AzureActuator{}
+
+// NewAzureActuator is the constructor for building a AzureActuator
+func NewAzureActuator(azureCreds *corev1.Secret, logger log.FieldLogger) (*AzureActuator, error) {
+	azureClient, err := azureclient.NewClientFromSecret(azureCreds)
+	if err != nil {
+		logger.WithError(err).Warn("failed to create Azure client with creds in clusterDeployment's secret")
+		return nil, err
+	}
+	actuator := &AzureActuator{
+		client: azureClient,
+		logger: logger,
+	}
+	return actuator, nil
+}
+
+// GenerateMachineSets satisfies the Actuator interface and will take a clusterDeployment and return a list of MachineSets
+// to sync to the remote cluster.
+func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *hivev1.MachinePool, logger log.FieldLogger) ([]*machineapi.MachineSet, error) {
+	if cd.Spec.ClusterMetadata == nil {
+		return nil, errors.New("ClusterDeployment does not have cluster metadata")
+	}
+	if cd.Spec.Platform.Azure == nil {
+		return nil, errors.New("ClusterDeployment is not for Azure")
+	}
+	if pool.Spec.Platform.Azure == nil {
+		return nil, errors.New("MachinePool is not for Azure")
+	}
+
+	ic := &installertypes.InstallConfig{
+		Platform: installertypes.Platform{
+			Azure: &installertypesazure.Platform{
+				Region: cd.Spec.Platform.Azure.Region,
+			},
+		},
+	}
+
+	computePool := baseMachinePool(pool)
+	computePool.Platform.Azure = &installertypesazure.MachinePool{
+		Zones:        pool.Spec.Platform.Azure.Zones,
+		InstanceType: pool.Spec.Platform.Azure.InstanceType,
+		OSDisk: installertypesazure.OSDisk{
+			DiskSizeGB: pool.Spec.Platform.Azure.OSDisk.DiskSizeGB,
+		},
+	}
+
+	if len(computePool.Platform.Azure.Zones) == 0 {
+		zones, err := a.getZones(cd.Spec.Platform.Azure.Region, pool.Spec.Platform.Azure.InstanceType)
+		if err != nil {
+			return nil, errors.Wrap(err, "compute pool not providing list of zones and failed to fetch list of zones")
+		}
+		if len(zones) == 0 {
+			return nil, fmt.Errorf("zero zones returned for region %s", cd.Spec.Platform.Azure.Region)
+		}
+		computePool.Platform.Azure.Zones = zones
+	}
+
+	// The imageID parameter is not used. The image is determined by the infraID.
+	const imageID = ""
+	// The role must be "worker" so that the machines get hooked up to the existing subnet.
+	const role = "worker"
+
+	installerMachineSets, err := installazure.MachineSets(cd.Spec.ClusterMetadata.InfraID, ic, computePool, imageID, role, "worker-user-data")
```

Comment:
worker-user-data should probably be treated the same as the other const's just above.